.github/workflows/performance.yaml

```diff
@@ -0,0 +1,82 @@
+#
+# When a PR is opened or updated which touches eBPF: run performance pipeline
+#
+name: Performance Pipeline
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "pkg/ebpf/c/**.c"
+      - "pkg/ebpf/c/**.h"
+      - "**/performance.yaml"
+permissions:
+  id-token: write
+  contents: read
+jobs:
+  publish-pr-image:
+    name: Publish PR Image
+    runs-on: [self-hosted, perf]
+    steps:
+      - name: AWS Authentication
+        uses: aws-actions/configure-aws-credentials@v4
+        env:
+          AWS_PROFILE: ${{ secrets.AWS_PROFILE }}
+        with:
+          aws-region: ${{ secrets.AWS_REGION}}
+          role-to-assume: ${{ secrets.AWS_RUNNER_ROLE_ARN }}
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Build Image and Publish
+        env:
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          REPO_USER: ${{ secrets.AWS_PR_IMAGE_ECR_REPO_USER }} 
+          REPO: ${{ secrets.AWS_ECR_NAME }}
+          PR_REPO: ${{ secrets.AWS_PR_IMAGE_ECR_REPO }}
+        run: |
+          make -f builder/Makefile.tracee-container build-tracee
+          aws ecr get-login-password --region $AWS_REGION | sudo docker login --username $REPO_USER --password-stdin $REPO
+          docker tag tracee:latest $PR_REPO:$GITHUB_SHA
+          docker push $PR_REPO:$GITHUB_SHA
+  network-benchmark:
+    name: Network Performance Gate
+    runs-on: [self-hosted, perf]
+    steps:
+      - name: AWS Authentication
+        uses: aws-actions/configure-aws-credentials@v4
+        env:
+          AWS_PROFILE: ${{ secrets.AWS_PROFILE }}
+        with:
+          aws-region: ${{ secrets.AWS_REGION}}
+          role-to-assume: ${{ secrets.AWS_RUNNER_ROLE_ARN }}
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Run Benchmark
+        env:
+          PR_REPO: ${{ secrets.AWS_PR_IMAGE_ECR_REPO }}
+        run: |
+          ./performance/benchmark/network/bench.sh $PR_REPO:$GITHUB_SHA
+      # Download previous benchmark result from cache (if exists)
+      - name: Download previous benchmark data
+        uses: actions/cache@v4.0.2
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1.20.1
+        with:
+          # What benchmark tool the output.txt came from
+          tool: 'smallerIsBetter'
+          # Where the output from the benchmark tool is stored
+          output-file-path: bench_output.json
+          # Where the previous data file is stored
+          external-data-json-path: ./cache/benchmark-data.json
+          # Workflow will fail when an alert happens
+          fail-on-alert: true
+          # Tag user when alerted
+          alert-comment-cc-users: '@NDStrahilevitz'
+          # Default value
+          alert-threshold: '200%'
```

Comment:
If our last latency ratio was 1.4 and after a PR it is 2.9, there will be an alert. But if its 1.6 there won't be. We may want to make this percent smaller to avoid drift.